File: apps/sdkwork-webserver-flutter-mobile/packages/sdkwork_webserver_flutter_mobile_applications/lib/src/routes/route_contributions.dart
/// Route contributions for the applications capability.
///
/// Route ids follow `<surface>.<domain>.<capability>.<screen>` and stay aligned
/// with the PC, H5, mini program, and HarmonyOS roots. Route metadata must not
/// declare HTTP API paths, SDK methods, or transport details.
///
/// The permission hint is the code the owning module derives from its own
/// operationId: `deploy.apps.list` -> `[resource, action] = ("apps", "list")` ->
/// `read` -> `deploy.apps.read`.
library;

import 'package:sdkwork_webserver_flutter_mobile_shell/sdkwork_webserver_flutter_mobile_shell.dart';

const List<WebserverFlutterRouteContribution>
webserverFlutterApplicationsRouteContributions =
    <WebserverFlutterRouteContribution>[
  WebserverFlutterRouteContribution(
    id: 'app.webserver.applications.list',
    domain: 'webserver',
    capability: 'applications',
    screen: 'list',
    routeName: '/applications',
    titleKey: 'applications.list.title',
    auth: WebserverFlutterRouteAuth.requiredAccess,
    permissionHint: 'deploy.apps.read',
    navigation: WebserverFlutterNavigationContribution(
      labelKey: 'navigation.applications',
      permission: 'deploy.apps.read',
      order: 10,
    ),
  ),
];

File: apps/sdkwork-webserver-mini-program/packages/sdkwork-webserver-mp-applications/src/routes/routeContributions.ts
import type { WebserverMpRouteContribution } from "@sdkwork/webserver-mp-shell";

/**
 * Route contributions for the applications capability.
 *
 * Route ids follow `<surface>.<domain>.<capability>.<screen>`
 * (`MINI_PROGRAM_APP_ARCHITECTURE_SPEC.md` §9). The physical path is a root-package
 * page because the console's first screen must render before any subpackage has
 * finished downloading; it must never declare an API URL or SDK method.
 *
 * `permissionHint` mirrors the PC console's `deploy.apps.read` entry, so both
 * clients gate the same screen on the same authority permission.
 */
export const webserverMpApplicationsRouteContributions: WebserverMpRouteContribution[] = [
  {
    id: "app.webserver.applications.list",
    surface: "app",
    domain: "webserver",
    capability: "applications",
    screen: "list",
    titleKey: "applications.list.title",
    auth: "required",
    permissionHint: "deploy.apps.read",
    navigation: {
      labelKey: "navigation.applications",
      permission: "deploy.apps.read",
      order: 10,
    },
    miniProgram: { rootPackage: true, pagePath: "pages/applications/index" },
  },
];
